src-tauri/src/domain/application.rs

```diff
@@ -1,82 +1,183 @@
+        let start_time =
+            get_process_start_time(pid).ok_or(TraceError::PIDNotFound { url: url.clone() })?;
+
+        Ok(Application {
+            pid,
             id: Uuid::new_v4(),
             title,
             url,
-            state: ApplicationState::Disabled,
-
+            start_time,
+            state: ApplicationState::Enabled,
+            connection_status: ConnectionStatus::Disconnected,
             connection: None,
-        }
+            cpu_usage: 0.0,
+            memory_usage: 0,
+        })
+    }
+
+    /// Current operating system PID of the running process.
+    pub fn pid(&self) -> u32 {
+        self.pid
+    }
+
+    /// Update the stored PID (for reattach scenarios).
+    pub fn set_pid(&mut self, pid: u32) {
+        debug!("Setting the pid to {}", pid);
+        self.pid = pid;
     }
 
+    /// Globally unique identifier for this application.
     pub fn id(&self) -> &Uuid {
         &self.id
     }
 
-    pub fn _title(&self) -> &str {
+    /// Title as provided by the user.
+    pub fn title(&self) -> &str {
         &self.title
     }
 
-    pub fn url(&self) -> &Url {
-        &self.url
+    /// The original URL endpoint used to connect.
+    pub fn url(&self) -> Url {
+        self.url.clone()
     }
 
+    /// Whether this application is enabled or disabled.
     pub fn state(&self) -> ApplicationState {
         self.state
     }
 
-    // vreau sa vad info pentru aplicatia asta
+    pub fn _cpu_usage(&self) -> f32 {
+        self.cpu_usage
+    }
+
+    /// Update the CPU‐usage statistic.
+    pub fn set_cpu_usage(&mut self, usage: f32) {
+        self.cpu_usage = usage;
+    }
+
+    pub fn _memory_usage(&self) -> u64 {
+        self.memory_usage
+    }
+
+    /// Update the memory‐usage statistic.
+    pub fn set_memory_usage(&mut self, usage: u64) {
+        self.memory_usage = usage;
+    }
+
+    pub fn _connection_status(&self) -> &ConnectionStatus {
+        &self.connection_status
+    }
+
+    /// Update the live connection status.
+    pub fn set_connection_status(&mut self, conn_status: ConnectionStatus) {
+        self.connection_status = conn_status;
+    }
+
+    /// Mark as enabled and stash the live connection object.
     pub fn enable(&mut self, connection: Connection) {
-        if self.state == ApplicationState::Disabled {
-            self.state = ApplicationState::Enabled;
-            self.connection = Some(connection);
-        }
+        self.state = ApplicationState::Enabled;
+        self.connection = Some(connection);
+        debug!("Stored connection");
     }
 
+    /// Mark as disabled and send a `Disconnect` command.
     pub async fn disable(&mut self) {
         if let Some(connection) = self.connection.take() {
             connection.commands.send(Command::Disconnect).await.ok();
             self.state = ApplicationState::Disabled;
         }
     }
+
+    // I need this test here in order for the `state_manager::state::tests::test_handle_and_get_pid`
```

Comment:
Preferinta personala aici, comentariile trebuie sa fie impersonale, la modul:
`This test is needed in order to ...`